### DISCUSSION TITLE
How to broadcast to live resource

### DISCUSSION THREAD

**Discussion by asok** [naymspace/backpex#495] [Q&A] [answered]
Hi, I'm new to elixir, but as far as I understand one should be able to use PubSub to send messages to live components, right?

Say I have an API and I create a new user, how can I broadcast the creation event so the live component of the live resource refreshes?
In live_resource.ex there's
```
      @impl Phoenix.LiveView
      def handle_info({"backpex:" <> unquote(event_prefix) <> "created", item}, socket)
          when socket.assigns.live_action in [:index, :resource_action] do
        {:noreply, refresh_items(socket)}
      end
```

I was hoping something like this might work:
```
    MyAppWeb.Endpoint.broadcast({"backpex:user_created", user})
```
But the function seems to accept more arguments.

**Comment by Flo0807** [comment, marked as answer]
Hey @asok 

You must broadcast the event using the same PubSub configuration as in your LiveResource.

Suppose your LiveResource configuration looks like this

```elixir
defmodule MyAppWeb.UserLive do
  use Backpex.LiveResource,
    # other configuration ...
    pubsub: MyApp.PubSub,
    topic: "users",
    event_prefix: "user_"
end
```

Your API can broadcast the creation of a user like this

```elixir
Phoenix.PubSub.broadcast(MyApp.PubSub, "users", {"backpex:user_created", item})
```

Backpex should now handle the event and update the UI accordingly.

**Reply by asok** [reply]
@Flo0807 thanks for answering I got it working after a small tweak. The correct invocation is:
```ex
Phoenix.PubSub.broadcast(MyApp.PubSub, "users", {"backpex:user_created", item})
```
The event name needs to be prefixed with "backpex:". Probably that comes from: https://github.com/naymspace/backpex/blob/develop/lib/backpex/live_resource.ex#L871

**Reply by Flo0807** [reply]
Nice! Yes, I forgot the `backpex:` prefix 😁 I updated my comment.
